src/hooks/chart.ts

```diff
@@ -230,21 +238,14 @@ export const useChartData = (plot: Plot) => {
    * TODO: create functions to retrieve only the neccesary templateParams from state and pass them into a more local scope
    *    - gene and study selectors touch these params to modify values used in API requests
    */
-  const templateParams: PlotTemplateParams = useMemo(
-    () => ({
-      $url_parameters: {
-        Gene: {
-          data: {
-            NCBI_GeneID: getQueryParam(windowRef.location.href, 'NCBI_GeneID') || 1, // TODO: deal with default value
-          },
-        },
-        Study: [],
-      },
-      $control_values: {},
-      noData: false, // TODO: remove hack when empty selectedRows are fixed
-    }),
-    []
-  );
+  const {templateParams,setTemplateParams, selectorOptionChanged, setSelectorOptionChanged} = useContext(TemplateParamsContext);
+
+  // To set the dataoptions for the dropdown controllers
+  useControl({
```

Comment:
This is being called in `plot-control-grid` too. I thought we discussed that the `useControl` in `plot-control-grid` was setting up global and local controls?

If that's the case, why is it repeated in here? Are the "local" controls actually being initialized here?